sdk/keyvault/keyvault-admin/src/accessControlModels.ts

```diff
@@ -20,6 +20,12 @@ export interface AccessControlClientOptions extends CommonClientOptions {
    * The accepted versions of the Key Vault's service API.
    */
   serviceVersion?: SUPPORTED_API_VERSIONS;
+
+  /**
+   * Whether to verify the authentication challenge resource matches the Key Vault or Managed HSM domain.
+   * Defaults to true.
```

Comment:
typically we avoid `boolean` options that default to true, since it's a tad confusing that something undefined is truthy.

Could we swap this to `disableChallengeResourceVerification` (or some better name) so it can default to `false`?

Reply:
It would be different from all other languages. The `verifyChallengeResource` was previously discussed/approved by some of the architects.

Reply:
I did run this API by Jose and Brian the other week and got the OK. But I don't think this issue (the property defaulting to true) came up, at least not explicitly.

/cc @joheredi @bterlson

Reply:
@heaths other languages are welcome to be on the correct side of boolean defaults too 😉

Reply:
We default to `true` in other cases, like with `DefaultAzureCredential` options, at least in .NET. Not sure what other languages look like, but properties to include (or not) credentials are `Include*` and some default to true while others false. That was the design inspiration here, in fact. You're welcome to be different, obviously, but other architects hadn't called this out.

Reply:
I don't feel very strongly and trust @timovv's judgement on this. I've just gotten bit by enough bugs caused by "not specified means true!" in my life that I make the effort to remark upon the danger now. 😄

Reply:
I am in the camp of an `disable/skip-` optional option to opt out of default behavior. Not strong opinion either if cross-language consistency is preferred.

Reply:
IMO, if there's no strong objection I'd recommend opting for consistency.

Reply:
In this case I think cross-language consistency wins out, given that it would be quite late in the picture to change the name. We are not expecting that the new option will get very much use.

Reply:
FWIW, Go may need to do some negation prefix. Let's see what @JeffreyRichter thinks.

Reply:
OK, if that's happening, I think it brings the balance back in favor of adding the negation.

Reply:
It is always best if Booleans default to false; this is good for all languages as I'm not aware of any language where you declare a Boolean variable/field and the language defaults it to true. - no customer would expect this.

Reply:
Renamed the property to `disableChallengeResourceVerification`.